src/bootstrap/flags.rs

```diff
@@ -614,10 +614,6 @@ Arguments:
         };
 
         if let Subcommand::Check { .. } = &cmd {
-            if matches.opt_str("stage").is_some() {
-                println!("--stage not supported for x.py check, always treated as stage 0");
-                process::exit(1);
-            }
             if matches.opt_str("keep-stage").is_some()
                 || matches.opt_str("keep-stage-std").is_some()
             {
```

Comment:
Why not remove the --keep-stage check too? e.g. if you modify the compiler docs or something but still want to check with stage 1.

Reply:
This PR doesn't support keep-stage, it would need to add in that support for copying artifacts into sysroots on keep-stage.

Reply:
Ok, can you change the error message to say that? Right now I think it says "check always uses stage 0" which is no longer the case.

Reply:
Done.